src/Symfony/Component/Console/Application.php

```diff
@@ -138,6 +138,9 @@ public function run(InputInterface $input = null, OutputInterface $output = null
                 $e = null;
                 $exitCode = 0;
```

Comment:
technically, the debug handler handles the exception (rendering it in a special way), but could still want to use a failure exit code. We should let the event change the code.